packages/backend/src/models/HuggingFaceModelHandler.ts

```
@@ -0,0 +1,143 @@
+  }
+  return undefined;
+}
+
+class HuggingFaceDownloader extends Downloader {
+  #target: string = '';
+
+  constructor(
+    url: string,
+    private repo: string,
+    private revision: string | undefined,
+    private abortSignal: AbortSignal,
+  ) {
+    super(url, '');
+  }
+
+  override getTarget(): string {
+    return this.#target;
+  }
+
+  async perform(id: string): Promise<void> {
+    const startTime = performance.now();
+
+    try {
+      this.#target = await snapshotDownload({
+        repo: this.repo,
+        revision: this.revision,
+      });
+      const durationSeconds = getDurationSecondsSince(startTime);
+      this._onEvent.fire({
+        id: id,
+        status: 'completed',
+        message: `Duration ${durationSeconds}s.`,
+        duration: durationSeconds,
+      } as CompletionEvent);
+    } catch (err: unknown) {
+      if (!this.abortSignal?.aborted) {
+        this._onEvent.fire({
+          id: id,
+          status: 'error',
+          message: `Something went wrong: ${String(err)}.`,
+        });
+      } else {
+        this._onEvent.fire({
+          id: id,
+          status: 'canceled',
+          message: `Request cancelled: ${String(err)}.`,
+        });
+      }
+      throw err;
+    } finally {
+      this.completed = true;
+    }
+  }
+}
+
+export class HuggingFaceModelHandler extends ModelHandler {
+  constructor(modelsManager: ModelsManager) {
+    super('huggingface model registry', modelsManager);
+  }
+
+  accept(url: string): boolean {
+    return url.startsWith('huggingface') || url.startsWith('hf');
+  }
+
+  createDownloader(model: ModelInfo, abortSignal: AbortSignal): Downloader {
+    const result = parseURL(model.url!);
+    if (result) {
+      return new HuggingFaceDownloader(model.url!, result.repo, result.revision, abortSignal);
+    }
+    throw new Error(`Invalid URL: ${model.url} for model ${model.name}`);
+  }
+
+  async deleteModel(model: ModelInfo): Promise<void> {
+    if (model.file) {
```

we should throw an error if we try to delete something that does not exists